compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -6566,6 +6566,48 @@ boolean isNeverTypeOrStructureTypeWithARequiredNeverMember(BType type, Set<BType
         }
     }
 
+    public boolean isContainClientObjectOrFunctionType(BType type, boolean isCheckForClient) {
+        Set<BType> visitedTypeSet = new HashSet<>();
+        visitedTypeSet.add(type);
+        return isContainClientObjectOrFunctionType(type, visitedTypeSet, isCheckForClient);
+    }
+
+    boolean isContainClientObjectOrFunctionType(BType type, Set<BType> visitedTypeSet, boolean isCheckForClient) {
+        if (type == null) {
+            return false;
+        }
+        type = Types.getReferredType(type);
+        if (isCheckForClient && Symbols.isFlagOn(type.tsymbol.flags, Flags.CLIENT)) {
+            return true;
+        }
+        switch (type.tag) {
+            case TypeTags.INVOKABLE:
+                return !isCheckForClient;
+            case UNION:
+                for (BType memberType: ((BUnionType) type).getMemberTypes()) {
+                    if (!visitedTypeSet.add(memberType)) {
+                        continue;
+                    }
+                    if (isContainClientObjectOrFunctionType(memberType, isCheckForClient)) {
+                        return true;
+                    }
```

Comment:
```suggestion
                    return isContainClientObjectOrFunctionType(memberType, isCheckForClient)
```

Reply:
imo, we cannot use this. It will break the for loop in the first iteration.
We want to iterate through the union member if we didn't found a `client object` or a `function type`.